programs/protocol_event/src/instructions/create_event.rs

```diff
@@ -50,6 +50,33 @@ fn validate_event(event_info: &CreateEventInfo) -> Result<()> {
         event_info.code.len() <= Event::MAX_CODE_LENGTH,
         EventError::MaxStringLengthExceeded,
     );
+
+    validate_event_timestamps(
+        event_info.expected_start_timestamp,
+        event_info.actual_start_timestamp,
+        event_info.actual_end_timestamp,
+    )?;
+
+    Ok(())
+}
+
+pub fn validate_event_timestamps(
+    expected_start_timestamp: i64,
+    actual_start_timestamp: Option<i64>,
+    actual_end_timestamp: Option<i64>,
+) -> Result<()> {
+    if let Some(actual_end_timestamp) = actual_end_timestamp {
+        require!(
+            expected_start_timestamp <= actual_end_timestamp,
+            EventError::InvalidTimestamp
+        );
+        if let Some(actual_start_timestamp) = actual_start_timestamp {
+            require!(
+                actual_start_timestamp <= actual_end_timestamp,
+                EventError::InvalidTimestamp
+            );
+        }
+    }
```

Comment:
Just an observation as the code seems correct, but the fact that it is testing `actual_end_timestamp` first kinda confused me as the timestamp relation works like `expected_start_timestamp <= actual_end_timestamp <= actual_end_timestamp` right?

Reply:
Wait ... `expected_start_timestamp <= actual_end_timestamp <= actual_end_timestamp` is not what is enforced.

So `expected_start_timestamp` can be later than `actual_start_timestamp`?

Reply:
there's no hard requirements tbh, but I wrote this with the intention that `expected_start` and `actual_start` don't happen in sequence, `actual_start` could be updated to be earlier than `expected_start`

So the validation is just checking that the end time is after both start times if it is set